sdk/keyvault/azure-security-keyvault-administration/inc/azure/keyvault/administration/rest_client_models.hpp

```diff
@@ -74,4 +74,238 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Administra
       std::vector<Setting> Value;
     };
 
+    /**
+     * @brief Define the options to create an SDK Keys client.
+     *
+     */
+    struct BackupRestoreClientOptions final : public Azure::Core::_internal::ClientOptions
+    {
+      /**
+       * @brief Service Version used.
+       *
+       */
+      const std::string ApiVersion{"7.5"};
+    };
+
+    /**
+     * @brief KeyVault Service Error model.
+     *
+     */
+    struct KeyVaultServiceError final
+    {
```

Comment:
Why is this exposed? This looks suspiciously like the ErrorCode and ErrorMessage fields in an Azure::Core::RequestFailedException.

Reply:
again .net has a similar pattern , they define the same object, with the same fields

Reply:
with one exception the request failed exception does not have an inner exception , which this one does.